gpiozero/output_devices.py

```diff
@@ -562,8 +562,12 @@ def __init__(
             raise GPIOPinMissing('red, green, and blue pins must be provided')
         LEDClass = PWMLED if pwm else LED
         super(RGBLED, self).__init__()
-        self._leds = tuple(LEDClass(pin, active_high) for pin in (red, green, blue))
-        self.value = initial_value
+        try:
+            self._leds = tuple(LEDClass(pin, active_high) for pin in (red, green, blue))
+            self.value = initial_value
+        except:
+            self.close()
+            raise
```

Comment:
...and also for ButtonBoard, SnowPi, FishDish, TrafficHat, Robot, _EnergenieMaster, Motor and Servo? (TBH I'm not sure exactly what the rules are as to when it is or isn't needed)